src/test/ui/associated-types/defaults-unsound-62211-1.stderr

```diff
@@ -55,6 +61,12 @@ LL | impl<T: std::fmt::Display> UncheckedCopy for T {}
 error[E0277]: the trait bound `T: std::ops::Deref` is not satisfied
   --> $DIR/defaults-unsound-62211-1.rs:41:9
    |
+LL | trait UncheckedCopy: Sized {
+   |       -------------
+...
+LL |     + Deref<Target = str>
+   |       ------------------- required by this bound in `UncheckedCopy`
```

Comment:
*really*, this bound is on the *associated type* -- I've been wanting to refactor the setup that we use for this for some time, actually...I guess orthogonal from this PR, but we could potentially detect it and underline the associated type. I'm imagining that we look for a bound whose self type is `<Self as Trait<...>>::Item` basically.

Reply:
This is true, but for the output to work correctly we'll need to collect more information in `BindingObligation`. I'd prefer to do so in a separate PR.

Reply:
Separate PR seems good.